AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_LightAdded.py

```diff
@@ -97,74 +182,299 @@ def AtomEditorComponents_Light_AddedToEntity():
-        general.idle_wait_frames(1)
-        Report.result(Tests.creation_redo, light_entity.exists())
+            # 3. Add Light component to the Light entity.
+            light_component = light_entity.add_component(AtomComponentProperties.light())
+            if iterations == 0:
+                Report.critical_result(Tests.light_component,
+                                       light_entity.has_component(AtomComponentProperties.light()))
 
-        # 5. Cycle through all light types.
-        for light_type in LIGHT_TYPES.keys():
+                # 4. UNDO the entity creation and component addition.
+                # -> UNDO component addition.
+                general.undo()
+                # -> UNDO naming entity.
+                general.undo()
+                # -> UNDO selecting entity.
+                general.undo()
+                # -> UNDO entity creation.
+                general.undo()
+                general.idle_wait_frames(1)
+                Report.result(Tests.creation_undo, not light_entity.exists())
+
+                # 5. REDO the entity creation and component addition.
+                # -> REDO entity creation.
+                general.redo()
+                # -> REDO selecting entity.
+                general.redo()
+                # -> REDO naming entity.
+                general.redo()
+                # -> REDO component addition.
+                general.redo()
+                general.idle_wait_frames(1)
+                Report.result(Tests.creation_redo, light_entity.exists())
+
+                # Update loop iterations to prevent additional running of these tests
+                iterations += 1
+
+            # 6. Set light type.
             light_component.set_component_property_value(
                 AtomComponentProperties.light('Light type'), LIGHT_TYPES[light_type])
             current_light_type = light_component.get_component_property_value(
                 AtomComponentProperties.light('Light type'))
             test_light_type = (
-                f"Light component has {light_type} type set",
-                f"Light component failed to set {light_type} type")
+                f"Set light type: {light_type.upper()}",
+                f"P0: Light component failed to set {light_type} type")
+            general.idle_wait_frames(1)
             Report.result(test_light_type, current_light_type == LIGHT_TYPES[light_type])
 
-        # 6. Enter/Exit game mode.
-        TestHelper.enter_game_mode(Tests.enter_game_mode)
-        general.idle_wait_frames(1)
-        TestHelper.exit_game_mode(Tests.exit_game_mode)
+            # Exit loop for unknown light type.
+            if current_light_type == 0:
+                continue
 
-        # 7. Test IsHidden.
-        light_entity.set_visibility_state(False)
-        Report.result(Tests.is_hidden, light_entity.is_hidden() is True)
+            # 7. Check for Shape component.
+            if current_light_type in (1, 2, 3, 4, 5):
```

Comment:
instead of numbering iterations and depending on order never changing
if light_type in LIGHT_SHAPES.keys():
the values 1, 2, 3, 4, 5 are dependent on the get_property_value call use light_type the key name of the light currently under test instead it should match the key name you define in your constant.

Reply:
changed to:
if light_type in ('sphere', 'spot_disk', 'capsule', 'quad', 'polygon'):